Yoast/wordpress-seo
# Add Wincher upgrade callouts

## Context

Add the Wincher account upgrade callouts in both edit-post and dashboard pages.

## Summary


This PR can be summarized in the following changelog entry:

- Add the Wincher upgrade callout to the Wincher performance report on the dashboard.
- Add the Wincher upgrade callout to the Wincher SEO performance on the edit post page.

## Test instructions

- Have Wincher integration enabled.
- Go to the edit post page.
- Add 4 keywords to Wincher i.e. add a keyphrase in "Focus keyphrase" and then enable tracking it in the "Track SEO performance" box.
- Check an upgrade callout is displayed above the keyphrases table.
- Go to the dashboard page.
- Check an upgrade callout is displayed above the keyphrases table, with a shortened title this time.

### Test instructions for the acceptance test before the PR gets merged

- An upgrade callout is displayed in edit-post and dashboard pages when the user has already used at least 80% of the account keyword limit (e.g. 4 keywords for the Yoast free account including 5 keywords)
- Upgrade title in the dashboard is shortened.
- If it's not a free account, we don't show "free" in the title on the edit-post page 
- We display the discount percentage and the number of months in the description if the Wincher campaign API returns a campaign with a "rate" type and months > 0.
- If no campaign is returned by the Wincher API, we don't display any discount information but we still display an upgrade callout with a link to the Wincher upgrade page.

#### Relevant test scenarios


### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

*

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

- The Wincher performance report on the dashboard page.
- The Wincher SEO performance box on the edit-post page.

## UI changes


## Other environments


## Documentation


## Quality assurance


## Innovation


## Evidence

- Months type campaign returned by the API, upgrade callout on the edit-post page:

![image](https://user-images.githubusercontent.com/129404933/235740598-5e5044bd-7d02-4b5a-878d-c8fd021e66ca.png)

- No campaign returned by the API: can't be tested without controlling the API

![image](https://user-images.githubusercontent.com/129404933/235740909-566c3c4b-08de-403e-9ed1-76383411c6e9.png)

- Amount type campaign: can't be tested without controlling the API

![image](https://user-images.githubusercontent.com/129404933/235741043-62677ae2-afc6-47c7-9675-835913d7b8c2.png)

- Upgrade callout on the dashboard page:

![image](https://user-images.githubusercontent.com/129404933/235741603-42221f57-23f4-4806-a271-5c7ffa3d50cc.png)
